ledger/participant-integration-api/src/main/scala/platform/store/dao/events/EventsTablePostgresql.scala

```diff
@@ -176,9 +176,19 @@ object EventsTablePostgresql extends EventsTable {
     val exerciseChildEventIds = "exerciseChildEventIds"
   }
 
-  private val insertStmt = {
+  /** Conflict detection relies on the `event_id` primary key only.
```

Comment:
This sounds a bit misleading.

If the indexer does perform post-commit validation (daml-on-sql), then it does so through [ActiveLedgerStateManager](https://github.com/digital-asset/daml/blob/main/ledger/participant-integration-api/src/main/scala/platform/store/ActiveLedgerStateManager.scala) and explicit SQL queries.

If the indexer does not perform post-commit validation, then we expect all writes to be free of conflicts because the underlying ReadService should not emit conflicting transactions.

All indices and constraints in the index database should be either performance optimizations or safeguards that are not strictly necessary.

Reply:
Ok, my aim intent was not clear then. I was referring to table-level insertion conflict (when inserting a row with an already-existing `event_id`) which is not a _update conflict_ for the pipelined indexer and should just be ignored. I've adapted the comment.